UniversalAlgebra/GaloisConnections/Properties.lagda

```diff
@@ -58,20 +58,26 @@ module _ {α ρ : Level}{𝒜 : Type α} where
          ; antisym = _,_
          }
 
+\end{code}
 
-module _ {ℓ ρ : Level}{𝒜 ℬ : Type ℓ} where
+A Binary relation from one poset to another induces a Galois connection, but only in a very special
+situation, namely when all the involved sets are of the same size.  This is akin to the situation
+with Adjunctions in Category Theory (unsurprisingly). In other words, there is likely a
+unit/counit definition that is more level polymorphic.
+\begin{code}
+module _ {ℓ : Level}{𝒜 : Type ℓ} {ℬ : Type ℓ} where
 
- 𝒫𝒜 𝒫ℬ : Poset (lsuc (ℓ ⊔ ρ)) (ℓ ⊔ ρ) (ℓ ⊔ ρ)
- 𝒫𝒜 = PosetOfSubsets{ℓ}{ρ}{𝒜 = 𝒜}
- 𝒫ℬ = PosetOfSubsets{ℓ}{ρ}{𝒜 = ℬ}
+ 𝒫𝒜 : Poset (lsuc ℓ) ℓ ℓ
+ 𝒫ℬ : Poset (lsuc ℓ) ℓ ℓ
+ 𝒫𝒜 = PosetOfSubsets ℓ 𝒜
+ 𝒫ℬ = PosetOfSubsets ℓ ℬ
 
  -- Every binary relation from one poset to another induces a Galois connection.
- Rel→Gal : (R : REL 𝒜 ℬ ρ) → Galois{ℓ}{ℓ}{ρ} 𝒫𝒜 𝒫ℬ
+ Rel→Gal : (R : REL 𝒜 ℬ ℓ) → Galois 𝒫𝒜 𝒫ℬ
```

Comment:
Okay, nice.  I thought I had managed to generalize a bit by using the level parameter ρ here... but I guess (according to your comment above) that's not possible...?

Reply:
The generalization was a bit of an illusion, as it forced all the previous definitions to be much less level-polymorphic than they could be - and much less level "natural", since everything was lifted to the least upper bound. What you want to do is to keep the levels as distinct as possible, unless you're defining a type, at which time you have no choice but to squash them all together.